### PR TITLE
Enable devserver proxy for custom environments

### PR DESCRIPTION
**Pull request by mchalandon**
It should be usefull to enable the devserver proxy for custom environements ; for projects with severall development environements.

### CODE REVIEW

File: lib/webpacker/dev_server.rb [hunk 1]
```diff
@@ -60,6 +60,6 @@ def fetch(key)
     end
 
     def defaults
-      config.send(:defaults)[:dev_server]
+      config.send(:defaults)[:dev_server] || {}
```
Comment (jasl) [inline lib/webpacker/dev_server.rb]: what about `config.dig(:defaults, :dev_server)`?

Comment (mchalandon) [reply lib/webpacker/dev_server.rb]: config is not a hash, but a Webpacker::Configuration which has a 'defaults' private method.

File: lib/webpacker/railtie.rb [hunk 2]
```diff
@@ -46,7 +46,9 @@ class Webpacker::Engine < ::Rails::Engine
   end
 
   initializer "webpacker.proxy" do |app|
-    if Webpacker.config.dev_server.present?
+    insert_middleware = Webpacker.config.dev_server.present? rescue nil
+    insert_middleware ||= false
+    if insert_middleware
```
Comment (jasl) [inline lib/webpacker/railtie.rb]: `nil` is equal to `false` in `if` statement
image

File: lib/webpacker/dev_server.rb (hunk 1 above)
Comment (jasl) [reply lib/webpacker/dev_server.rb]: oh i see

File: lib/webpacker/railtie.rb (hunk 2 above)
Comment (mchalandon) [reply lib/webpacker/railtie.rb]: you're right ! thanks 
ed6743a170242e7611795a21b9581b7992a6d061

**Pull request merged**